benchmarks/advection_stabilization/advection_stabilization.cc

```diff
@@ -0,0 +1,170 @@
+#include <aspect/boundary_velocity/interface.h>
+#include <aspect/simulator_access.h>
+#include <aspect/global.h>
+#include <aspect/gravity_model/interface.h>
+#include <aspect/postprocess/dynamic_topography.h>
+
+#include <aspect/prescribed_stokes_solution/interface.h>
+#include <aspect/simulator_access.h>
+
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/numerics/data_out.h>
+#include <deal.II/base/quadrature_lib.h>
+#include <deal.II/base/function_lib.h>
+#include <deal.II/numerics/error_estimator.h>
+#include <deal.II/numerics/vector_tools.h>
+
+namespace aspect
+{
+  /**
+   * This is a variation of the "annulus" benchmark in benchmark/annulus
+   * that tests the influence of the stabilization method on the temperature
+   * equation.
+   *
+   */
+  namespace AnnulusBenchmark
+  {
+    using namespace dealii;
+
+    namespace AnalyticSolutions
+    {
+      const double A=2.0, B=-3.0/std::log(2.0), C=-1;
+      const double outer_radius = 2.;
+      const double rho_0 = 1000.;
+      const double gravity = 1.;
+
+      Tensor<1,2>
+      Annulus_velocity (const Point<2> &pos,
+                        const double k)
+      {
+        const double x = pos[0];
+        const double y = pos[1];
+        const double r = std::sqrt(x*x+y*y);
+        const double theta = std::atan2(y,x);
+        const double f_r = A*r + B/r;
+        const double g_r = A*r/2 + B*std::log(r)/r + C/r;
+        const double v_r = g_r*k*sin(k*theta);
+        const double v_theta = f_r*cos(k*theta);
+        const double v_x = cos(theta)*v_r - sin(theta)*v_theta;
+        const double v_y = sin(theta)*v_r + cos(theta)*v_theta;
+        return Point<2> (v_x,v_y);
+      }
+
+      double
+      Annulus_pressure (const Point<2> &pos,
+                        const double k)
+      {
+        const double x = pos[0];
+        const double y = pos[1];
+        const double r=std::sqrt(x*x+y*y);
+        const double theta = std::atan2(y,x);
+        const double f_r = 2*r + B/r;
+        const double g_r = A*r/2 + B*std::log(r)/r + C/r;
+        const double h_r=(2*g_r-f_r)/r;
+        return k*h_r*sin(k*theta)+rho_0*gravity*(outer_radius-r);
+      }
+
+      template<int dim>
+      double
+      Annulus_normal_traction (const Point<dim> &pos,
+                               const double k)
+      {
+        Assert (dim == 2, ExcNotImplemented());
+        const double x = pos[0];
+        const double y = pos[1];
+        const double r=std::sqrt(x*x+y*y);
+        const double theta = std::atan2(y,x);
+        const double f_r = 2*r + B/r;
+        const double g_r = A*r/2 + B*std::log(r)/r + C/r;
+        return k * 3.*f_r/r * std::sin(k*theta) - rho_0 * g_r * (outer_radius - r);
+      }
+
+    }
+  }
+}
+
+
+namespace aspect
+{
+  namespace PrescribedStokesSolution
+  {
+    using namespace dealii;
+
+    /**
+     * A class that implements a circular, divergence-free flow field
+     * around the origin of the coordinate system.
```

Comment:
Please update this comment to say that this class implemented the flow field of the annulus benchmark